src/modeler/AutoRest.Swagger/Validation/ListByOperationsValidation.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class ListByOperationsValidation : TypedRule<Dictionary<string, Dictionary<string, Operation>>>
+    {
+        private readonly Regex ListByRegex = new Regex(@".+_listby.+$", RegexOptions.IgnoreCase);
+        private readonly Regex UrlRegex = new Regex(@"(\/[^\/]+)+$", RegexOptions.IgnoreCase);
```

Comment:
use named regex groups like `(?<foo>bar)`! Then you can go with `match.Groups["foo"]` rather than `match.Groups[MAGIC_NUMBER]` (note that the magic number can change when the regex is changed so suddenly hell breaks loose tracking all the required changes down)